base/linalg/tridiag.jl

```diff
@@ -276,10 +278,24 @@ function Tridiagonal{T}(dl::Vector{T}, d::Vector{T}, du::Vector{T})
     end
     Tridiagonal(dl, d, du, zeros(T,n-2))
 end
-function Tridiagonal{Tl, Td, Tu}(dl::Vector{Tl}, d::Vector{Td}, du::Vector{Tu})
+
+# Construct from diagonals of any abstract vector, any eltype
+function Tridiagonal{Tl, Td, Tu}(dl::AbstractVector{Tl}, d::AbstractVector{Td}, du::AbstractVector{Tu})
     Tridiagonal(map(v->convert(Vector{promote_type(Tl,Td,Tu)}, v), (dl, d, du))...)
 end
 
+# Provide a constructor Tridiagonal(A) similar to the triangulars, diagonal, symmetric
+"""
+    Tridiagonal(A)
+
+returns a `Tridiagonal` array based on (abstract) matrix `A`, using its lower diagonal,
+diagonal, and upper diagonal. This applies even for non-square `A`, but
+yielding a square `Tridiagonal` matrix.
```

Comment:
Oops, that was for an earlier version. I ditched that as a silly feature and forgot to update the docstring. I've removed the incorrect part of the docstring.